tests/testutil.hpp

```diff
@@ -30,11 +30,7 @@
 #ifndef __TESTUTIL_HPP_INCLUDED__
 #define __TESTUTIL_HPP_INCLUDED__
 
-#if defined ZMQ_CUSTOM_PLATFORM_HPP
```

Comment:
why is this being removed?

Reply:
I removed it, because visual code intellisense was complaining about the platform header file, as it doesn't exist. Cmake will generate it into the cmake build directory, but not in ../src/platform.hpp. I did not go back in time. Perhaps it was used in the deprecated msvc solutions,  but it is not being used in the current cmake build environment (for windows). 

I indeed did not finish this completely, and I should remove it everywhere.  I'll give it a try.

NB windows x64 test_security_zap has an attitude. It might need some attention.

Reply:
Autotools will generate it in src. Please remove this change and fix your edit settings instead.

Reply:
Isn't the #ifdef ZMQ_CUSTOM_PLATFORM_HPP logic only used in cmake / windows ?
I didn't find it anywhere else in the repo, The autotools builds should fail miserably if they were missing the header file.

Let's wait on the build results. 
Maybe I should have created a separate PR for this.

Reply:
No, this is wrong - autotools always generates in src. Please remove it.

Reply:
The builds and test runs seem fine to me.
There is no problem with autotools generating the platform.hpp in src directory.
There is just no need to use the #ifdef  ZMQ_CUSTOM_PLATFORM_HPP  logic

I'm happy to put it back, upon your request, though I don't understand what is going wrong here with the builds.